src/DataMapper/EntityAssembler.php

```diff
@@ -25,18 +25,16 @@ class EntityAssembler
      *
      * @param array|object $document the attributes that will be used to compose the entity
      * @param Schema       $schema   schema that will be used to map each field
-     *
-     * @return mixed
      */
-    public function assemble($document, Schema $schema)
+    public function assemble(array|object $document, Schema $schema): mixed
```

Comment:
```suggestion
    public function assemble(array $document, Schema $schema): mixed
```
Isso aqui só é usado ali no `foreach` então suponho que seja só um array

Reply:
Aqui eu alterei, mas existe um teste que valida o envio de um object ali, então imagino que seja uma funcionalidade receber isso